application/browser/application_system_tizen.cc

```diff
@@ -59,23 +60,40 @@ namespace {
 
 void application_event_cb(app_event event, void* data, bundle* b) {
   LOG(INFO) << "Received Tizen appcore event: " << event;
-  ApplicationTizen* app = reinterpret_cast<ApplicationTizen*>(data);
-  CHECK(app);
+  ApplicationSystemTizen* app_system =
+      reinterpret_cast<ApplicationSystemTizen*>(data);
+  CHECK(app_system);
 
   switch (event) {
     case AE_UNKNOWN:
     case AE_CREATE:
       break;
     case AE_TERMINATE:
-      app->Terminate();
+      if (app_system->current_app())
+        app_system->current_app()->Terminate();
       break;
     case AE_PAUSE:
-      app->Suspend();
+      if (app_system->current_app())
+        app_system->current_app()->Suspend();
       break;
     case AE_RESUME:
-      app->Resume();
+      if (app_system->current_app())
+        app_system->current_app()->Resume();
       break;
-    case AE_RESET:
+    case AE_RESET: {
+      base::FilePath exec_path(app_system->launch_params()->argv()[0]);
+      std::string app_id = exec_path.BaseName().MaybeAsASCII();
+      LOG(INFO) << "Attempting to launch the app with id: " << app_id;
+      if (!IsValidApplicationID(app_id))
+        break;
+
+      ApplicationServiceTizen* app_service_tizen =
+          ToApplicationServiceTizen(app_system->application_service());
+      Application* app = app_service_tizen->LaunchFromAppID(app_id);
```

Comment:
What do you exactly mean when saying "re-assign"? I need ApplicationSystem in this handler, not Application. It is not possible to reassign date for handler.